File: src/B11723.c
#include <stdio.h>
#include <string.h>
int arr[20] = {0,};
void add(int x){
  if(arr[x] == 1)
    return;
  else
    arr[x] = 1;
}
void rem(int x){
  if(arr[x] == 1)
    arr[x] = 0;
  else
    return;
}
void check(int x){
  if (arr[x] == 1) 
    printf("1\n");
  else
    printf("0\n");
}
void toggle(int x){
  if (arr[x] == 1)
    rem(x);
  else
    add(x);
}
void all(){
  for(int i=1; i<=20; i++){
    arr[i] = 1;
  }
}
void empty(){
  for(int i=1; i<=20; i++){
    arr[i] = 0;
  }
}
int main(void){
  int n, x;
  scanf("%d", &n);
  char s[10];
  for(int i=0; i<n; i++){
    scanf("%s %d", s, &x);
    if (!strcmp(s, "add"))
      add(x);
    else if(!strcmp(s, "remove"))
      rem(x);
    else if(!strcmp(s, "check"))
      check(x);
    else if(!strcmp(s, "toggle"))
      toggle(x);
    else if(!strcmp(s, "all"))
      all();
    else if(!strcmp(s, "empty"))
      empty();
  }
  return 0;
}

File: src/Bit Masking/B11723.py
import sys
input = sys.stdin.readline

M = int(input())
S = 0
for _ in range(M) :
    command = input().split()
    op = command[0]
    if len(command) > 1 :
        num = int(command[1])
        if op == "add" :
            S |= (1 << num)
        elif op == "remove" :
            S &= ~(1 << num)
        elif op == "check" :
            print(1 if S & (1 << num) else 0)
        elif op == "toggle" :
            if S & (1 << num) :
                S &= ~(1 << num)
            else :
                S |= (1 << num)
    else :
        if op == "all" :
            S = (1 << 21) - 1
        elif op == "empty" :
            S = 0